pygeoapi/api.py

```diff
@@ -1429,6 +1452,39 @@ def get_collection_items(
                 HTTPStatus.INTERNAL_SERVER_ERROR, headers, request.format,
                 'NoApplicableCode', msg)
 
+        LOGGER.debug('Processing bbox-crs parameter')
+        bbox_crs = request.params.get('bbox-crs')
+        if bbox_crs is not None:
+            if len(bbox) == 0:
+                msg = 'bbox-crs specified without bbox parameter'
+                return self.get_exception(
+                    HTTPStatus.BAD_REQUEST, headers, request.format,
+                    'NoApplicableCode', msg)
+
+            if len(bbox_crs) == 0:
+                msg = 'bbox-crs specified but is empty'
+                return self.get_exception(
+                    HTTPStatus.BAD_REQUEST, headers, request.format,
+                    'NoApplicableCode', msg)
+
+            supported_crs_list = provider_def.get('crs', DEFAULT_CRS_LIST)
```

Comment:
I think that the `supported_crs_list` should include all *CRSs* from `DEFAULT_CRS_LIST`, even if they are not include in the config for the provider (`crs` field). I have included them in https://github.com/geopython/pygeoapi/pull/1149 with:

```python

supported_crs_list = provider_def.get('crs', list())
for uri in DEFAULT_CRS:
    if uri not in supported_crs_list:
        supported_crs_list.append(uri)
```

But as this list of supported CRSs is used for processing both the `bbox-crs` and `crs` parameter, maybe we should move these lines up and out of these `if` *statetements* for re-use?

Reply:
A left-over is also to deal with http://www.opengis.net/def/crs/OGC/0/CRS84h (for coordinates with ellipsoidal height). The better to have a single check/append.

Reply:
Yes. According to the standards, the default CRS is either [http://www.opengis.net/def/crs/OGC/0/CRS84](http://www.opengis.net/def/crs/OGC/0/CRS84) (coordinates without height) or [http://www.opengis.net/def/crs/OGC/0/CRS84h](http://www.opengis.net/def/crs/OGC/0/CRS84h) (coordinates with ellipsoidal height). But to know that, one should get at least one `item`/`Feature` from the collection, and see if it has 2D/3D coordinates... Is sending one request through `get_collection_item` too expensive (maybe with `&limit=1`)? The returned item must also have a non-`NULL` geometry...

Reply:
Yes that is expensive, at least cumbersome. My suggestion is that in that case one should configure explicitly.

Ha: It appears there are two versions of the "Part 2" standard
https://docs.ogc.org/is/18-058/18-058.html and
https://docs.opengeospatial.org/is/18-058r1/18-058r1.html

Diff seems only "editorial" though, comparing through
https://www.copyscape.com/compare.php

Version 18-058r1 van 2022-05-11 is listed on https://www.ogc.org/standards/ so that is the "official one".

Maybe I was looking at intermediate versions as I thought the default CRSs for the Collection should be:
    http://www.opengis.net/def/crs/EPSG/0/4326
and one of
    http://www.opengis.net/def/crs/OGC/1.3/CRS84 or
    http://www.opengis.net/def/crs/OGC/1.3/CRS84h

But can't find that (http://www.opengis.net/def/crs/EPSG/0/4326) now, looks like the default CRS 'list' is:
one of
    http://www.opengis.net/def/crs/OGC/1.3/CRS84 or
    http://www.opengis.net/def/crs/OGC/1.3/CRS84h

Which makes sense as these have lon, lat axis ordering like GeoJSON default requires..

Reply:
Yes, http://www.opengis.net/def/crs/EPSG/0/4326 is not part of the default CRSs in the standard.

Something I have also thought about, is how pygeoapi handles the coordinates of the `bbox` query parameter. In the part 1 of the standard, [requirement 23 B](https://docs.ogc.org/is/17-069r4/17-069r4.html#_parameter_bbox) use the wording *"coordinate axis1, coordinate axis 2"*. Axis 1 for http://www.opengis.net/def/crs/EPSG/0/4326 is for latitude, while the Axis 1 is longitude for http://www.opengis.net/def/crs/OGC/1.3/CRS84 . I think pygeoapi handles a `bbox` as `min_x, min_y[, min_z], max_x, max_y[, max_z]` even if `bbox-crs` will be set to http://www.opengis.net/def/crs/EPSG/0/4326

Reply:
Cannot tell if/how pygeoapi  bbox-handling affects axis ordering. At least it checks if `min_x < max_x` etc, but that is not affected by ordering IMHO. I'll be writing more tests, also with http://www.opengis.net/def/crs/EPSG/0/4326, so that may come out.

I have just changed the "default CRS" handling what I think is according to the Part 2 standard. Having the util-function came in handy!

Reply:
Well, maybe not when handling the `bbox` itself. Just wondering how pygeoapi deals with the different providers (and data sources) that may use different conventions regarding the axis order lat/lon or lon/lat.

Nice work!